### ISSUE TITLE
boundAlloc uses LoadLibraryA to load kernelbase.dll and never frees it after

### ISSUE THREAD

**Issue by lkalam** [stevemk14ebr/PolyHook_2_0#207]
I am not 100% sure this is an issue, but I've noticed that LoadLibraryA("kernelbase.dll") is called in boundAlloc() and boundedAllocSupported() but the respective FreeLibrary(hMod) is never called to reduce the reference number to kernelbase.dll and/or free it eventually...

**Comment by stevemk14ebr** [comment]
Not an issue kernel32 is going to be loaded into every process already, except some weird esoteric cases where it would be freed in process shutdown anyways. Load library being used in this way is more to get a handle to the module than to trigger a load

**Comment by lkalam** [comment]
Fair enough - just for my own curiosity though, why not use GetModuleHandle() for it instead?

**Comment by stevemk14ebr** [comment]
Because if it's not loaded I still need it. There would be a leak you are right but it's handled by the process stopping and kernel32 is normal enough to have hanging around

**Comment by lkalam** [comment]
so - if you tried to get the handle using GetModuleHandle() and if that returns NULL, use LoadLibrary() for it?

Sorry if I am misunderstanding something here... but boundAlloc() can be called multiple times when hooking / unhooking etc. so it seems that there is a potential for resource leaks (according to ChatGPT that is).

**Comment by stevemk14ebr** [comment]
First off, chatgot doesn't know shit, ask a human.

Load library returns a handle to the module if it's already loaded, if it's not then it loads it and returns the handle. It will not load the module multiple times.

**Comment by lkalam** [comment]
First off, no need to get testy - I did check the documentation which does state that LoadLibrary increments a ref count and assorted other stuff which adds some load to the functionality.

I did ask several humans, the point is, if a DLL is already loaded in memory, the proper way to access its functions is through checking what GetModuleHandle() returns first, then if it returns NULL, use LoadLibrary ONCE and keep a reference handle to it (optionally of course, if you don't want to keep reusing GetModuleHandle()).

Using LoadMemory() a thousand times is not the proper way to do it.

But, your code, your call. Leave it as-is if you like.
